openhexa/sdk/pipelines/parameter.py

```diff
@@ -307,6 +308,34 @@ def to_connection(self, value: str) -> CustomConnection:
         return workspace.custom_connection(value)
 
 
+class DatasetType(ParameterType):
+    """Type class for dataset parameter."""
+
+    @property
+    def spec_type(self) -> str:
+        """Return a type string for the specs that are sent to the backend."""
+        return "dataset"
+
+    @property
+    def expected_type(self) -> type:
+        """Returns the python type expected for values."""
+        return DatasetType
+
+    def validate(self, value: typing.Optional[typing.Any]) -> Dataset:
+        """Validate the provided value for this type."""
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        try:
+            return self.to_dataset(value)
+        except ValueError as e:
+            raise ParameterValueError(str(e))
+
+    def to_dataset(self, value: str) -> CustomConnection:
```

Comment:
This method is not needed. We had `to_connection` for connection parameter types because they share the same `validate()` method. Here, you can just adapt `validate()` so that it return the dataset.

Reply:
All right then.